src/Microsoft.Windows.ComputeVirtualization/Container.cs

```diff
@@ -134,18 +174,24 @@ public Process CreateProcess(ProcessStartInfo startInfo)
         /// <returns></returns>
         public Process GetProcess(int pid)
         {
-            return new Process(_id, pid, null, null, null, false);
+            IntPtr process = new IntPtr();
+            string result;
+            HcsFunctions.ProcessHcsCall(HcsFunctions.HcsOpenProcess(_cs, (uint)pid, out process, out result));
+            return new Process(process, null, null, null, false);
         }
 
         /// <summary>
-        /// Disposes of the container object, killing the container if <see cref="ContainerSettings.KillOnClose"/> was set.
+        /// Disposes of the container object.
         /// </summary>
         public void Dispose()
         {
-            if (_killOnClose)
+            if (_killOnClose && !_killed)
             {
                 Kill();
             }
+
+            _watcher.Dispose();
+            HcsFunctions.ProcessHcsCall(HcsFunctions.HcsCloseComputeSystem(_cs));
```

Comment:
I think so, but I'll have to double check with @darrenstahlmsft if there is some well known error I should look out for here.